compiler/luci/service/src/Validate.cpp

```diff
@@ -107,10 +107,27 @@ bool validate_shape_dtype(loco::Graph *g)
     auto go_tensor_shape = graph_out->shape();
     assert(go_tensor_shape);
 
+    // NOTE Even if shape of graph output is [] (which means "shape inference was impossible")
+    //      but shape of CircleNode is not, it can be valid case because shape inference
+    //      algorithm of CircleNode may be upgraded than before. The opposite is possible either.
+    //      If such cases are appeared, following validation code should be fixed.
     bool is_shape_valid = (circle_node->rank() == go_tensor_shape->rank());
     for (uint32_t i = 0; is_shape_valid && i < circle_node->rank(); ++i)
+    {
       if (circle_node->dim(i).value() != go_tensor_shape->dim(i).value())
-        is_shape_valid = false;
+      {
+        if (!circle_node->dim(i).known() || !go_tensor_shape->dim(i).known())
```

Comment:
Just curiosity, is it normal to check if dim is known before comparing its value?

Reply:
Logic has no problem but it may cause unintended confusion, I will fix it :)